src/Symfony/Component/VarDumper/Resources/functions/dump.php

```diff
@@ -11,6 +11,11 @@
 
 use Symfony\Component\VarDumper\VarDumper;
 
+// Register & lock the handler with expected format as soon as the env var was set:
+if (isset($_SERVER['VAR_DUMPER_FORMAT'])) {
```

Comment:
this is too early for being compatible with `.env`...

Reply:
You're right....but I really didn't have this use-case in mind.

My opinion here is that using the `VAR_DUMPER_FORMAT` is not a `.env` file use-case at all. You'll use it in CLI to get the single format you expected, whatever the stack does behind. We need this to be registered as soon as possible to ensure the format is locked and cannot be replaced by the stack. Otherwise `setHandler` can be called before the very first dump and take precedence over expected format. 

So...what about making `setHandler` a noop instead as soon as the `VAR_DUMPER_FORMAT` var is set (removing the lock concept by the same way)?